reader.go

```diff
@@ -19,33 +21,104 @@ type ParseError struct {
 	Err    error  // The actual error
 }
 
-func (e *ParseError) Error() string {
+func (e ParseError) Error() string {
 	if e.Record == "" {
 		return fmt.Sprintf("line:%d %T %s", e.Line, e.Err, e.Err)
 	}
 	return fmt.Sprintf("line:%d record:%s %T %s", e.Line, e.Record, e.Err, e.Err)
 }
 
+// ErrorList represents an array of errors which is also an error itself.
+type ErrorList []error
+
+// Add appends err onto the ErrorList. Errors are kept in order.
+func (e *ErrorList) Add(err error) {
+	*e = append(*e, err)
+}
+
+// Err returns the first error (or nil).
+func (e ErrorList) Err() error {
+	if e == nil || len(e) == 0 {
+		return nil
+	}
+	return e[0]
+}
+
+// Error implements the error interface
+func (e ErrorList) Error() string {
+	if len(e) == 0 {
+		return "<nil>"
+	}
+	var buf bytes.Buffer
+	e.Print(&buf)
+	return buf.String()
+}
+
+// Print formats the ErrorList into a string written to w.
+// If ErrorList contains multiple errors those after the first
+// are indented.
+func (e ErrorList) Print(w io.Writer) {
+	if w == nil || len(e) == 0 {
+		fmt.Fprintf(w, "<nil>")
+		return
+	}
+
+	fmt.Fprintf(w, "%s", e[0])
+	if len(e) > 1 {
+		fmt.Fprintf(w, "\n")
+	}
+
+	for i := 1; i < len(e); i++ {
+		fmt.Fprintf(w, "  %s", e[i])
+		if i < len(e)-1 { // don't add \n to last error
+			fmt.Fprintf(w, "\n")
+		}
+	}
+}
+
+func (e ErrorList) Empty() bool {
+	return e == nil || len(e) == 0
+}
+
+func (e ErrorList) MarshalJSON() ([]byte, error) {
```

Comment:
Should we render this in the endpoints as an array of errors? JSON arrays are ordered (per the spec), but I've seen clients not respect that.

Reply:
I think so, Wade?  If so should we document that we are following the spec?

Reply:
I'm tempted to just return a string (that has no format). I'd rather not go down the rabbit trail of string parsing error messages.

Reply:
Are you saying you would return a string which contained all errors? If so,  how would they be distinguished for a client?

Reply:
Right now it's a big string, with newlines and an indent.

```
first error
  second error
  ...
```

Reply:
There's a test showing the format too: https://github.com/moov-io/ach/pull/299/files#diff-f86b763600291cbceee077a33133434aR53

Reply:
The spec is a suggestion on an implementation. We are free to change the specification however we see fit. I agree with @bkmoovio that the error will probably just be logged.